### DISCUSSION TITLE
Error compiling project using vector_graphics_compiler for SVG (Android)

### DISCUSSION THREAD

**Discussion by fvisticot** [codemagic-ci-cd/codemagic-docs#2629] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I need to rebuild multiple time the workflow on CodeMagic to get it works (sometimes 5 times).
When it fails, I get this error:

pubspec.yaml
```
dev_dependencies:
  vector_graphics: ^1.1.11+1

  assets:
    - assets/
    - assets/fonts/
    - assets/pngs/
    - path: assets/svgs/
      transformers:
        - package: vector_graphics_compiler
```

### Expected results

Compilation OK

### Actual results

Target aot_android_asset_bundle failed: Error: User-defined transformation of asset "/Users/builder/clone/assets/svgs/logo-tibib.svg" failed.
Transformer process terminated with non-zero exit code: 255
Transformer package: vector_graphics_compiler
Full command: /Users/builder/programs/flutter/bin/cache/dart-sdk/bin/dart run vector_graphics_compiler --input=/var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/flutter_tools.aB9YxI/logo-tibib.svg-transformOutput0.svg --output=/var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/flutter_tools.aB9YxI/logo-tibib.svg-transformOutput1.svg
stdout:

stderr:
The ".dart_tool/package_config.json" file is not recognized by "pub" version, please run "dart pub get".
The ".dart_tool/package_config.json" file is not recognized by "pub" version, please run "dart pub get".

**Comment by himesh-cm** [comment]
Hey @fvisticot, Can you please share the build ID with me? You can find it under your build details page's 'ID' section.  ﻿Navigate to the 'Builds' section on the left-hand menu, select the build you want to view, and click the copy icon.

**Comment by fvisticot** [comment]
Hello, 
Tx you for your quick replay:
Build ID: 668863a2e3ee9165e5804d47

When I remove the generate: true from pubscpec.yaml it works fine

Regards

**Reply by himesh-cm** [reply]
Hey @fvisticot, what flutter version do you use locally, can you confirm if you are using the same version on codemagic as well?

**Reply by himesh-cm** [reply]
Hey @fvisticot, Did you get a chance to try it as per my last suggestion?

**Comment by himesh-cm** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.

**Comment by corepuncher** [comment]
Hey all, did anyone fix this?  I cannot get my items to compile.  It acts like it works on most then fails in the end if I have over 5 svg's in a dir:

Here is my test results:
 
--------------------
1) In pubspec.yaml, only tried transform on a single dir with 31 svgs, each 200 to 500 bytes in size.

```
    - path: assets/xyz/
      transformers:
        - package: vector_graphics_compiler
```

(flutter run)

FAILED

--------------------
2) Made new dir /test with 1 svg item in it.

```
    - path: assets/test/
      transformers:
        - package: vector_graphics_compiler
```

SUCCESS

--------------------
3) Copied the 31 svg's from /xyz into the /test dir

FAILED

--------------------
4) Went back and tried the single svg item again

SUCCESS

--------------------
5) This time tried with 2 svg's in /test dir

SUCCESS

--------------------
6) Tried with 15 svg's in /test dir

FAILED

It transformed 11 items: 

```
[+1999 ms] [+2082 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item30.svg' (2406ms)
[  +94 ms] [  +33 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item45.svg' (2437ms)
[        ] [  +18 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item65.svg' (2454ms)
[        ] [  +23 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item40.svg' (2481ms)
[        ] [   +6 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item50.svg' (2483ms)
[        ] [   +1 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item20.svg' (2491ms)
[  +93 ms] [  +36 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item55.svg' (2523ms)
[        ] [   +2 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item10.svg' (2534ms)
[        ] [   +3 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item60.svg' (2529ms)
[        ] [   +1 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item35.svg' (2533ms)
[        ] [  +12 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item15.svg' (2548ms)

Stuck for several minutes (CPU loud), then

[+341702 ms] [+341688 ms] Persisting file store
[   +2 ms] [  +18 ms] Done persisting file store
[   +2 ms] [  +11 ms] Target debug_android_application failed: Error: User-defined transformation of asset 
"C:\Users\me\projectname\dev-projectname\assets\test\item70.svg" failed.
[   +3 ms]            Transformer process terminated with non-zero exit code: 255
[        ]            Transformer package: vector_graphics_compiler
[        ]            Full command: C:\Flutter\bin\cache\dart-sdk\bin\dart.exe run vector_graphics_compiler
--input=C:\Users\me\AppData\Local\Temp\flutter_tools.6ccc28c2\item70.svg-transformOutput0.svg
--output=C:\Users\me\AppData\Local\Temp\flutter_tools.6ccc28c2\item70.svg-transformOutput1.svg
[        ]            stdout:
[        ]            stderr:
[        ]            Cannot rename file to
'C:\Users\me\projectname\dev-projectname\.dart_tool/pub\bin\vector_graphics_compiler\vector_graphics_compile
r.dart-3.5.3.snapshot', path =
'C:\Users\me\projectname\dev-projectname\.dart_tool/pub\bin\vector_graphics_compiler\tmp449ac982\vector_grap
hics_compiler.dart-3.5.3.snapshot.incremental.temp' (OS Error: Access is denied.
[        ]            , errno = 5)
[        ]            PathAccessException: Cannot rename file to
'C:\Users\me\projectname\dev-projectname\.dart_tool/pub\bin\vector_graphics_compiler\vector_graphics_compile
r.dart-3.5.3.snapshot', path =
'C:\Users\me\projectname\dev-projectname\.dart_tool/pub\bin\vector_graphics_compiler\tmp449ac982\vector_grap
hics_compiler.dart-3.5.3.snapshot.incremental.temp' (OS Error: Access is denied.
[        ]            , errno = 5)
[        ]            #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
[        ]            #1      copyAssets.<anonymous closure>
...dart errors
```


--------------------
7) This time I limited it to 10 svg's, and ensured I included item70.svg

FAILED

It transformed 9 of the items, INCLUDING item70.svg.  This time, it "failed" at item 65.svg with same errors as above.

--------------------
8) This time, tried 5 svg's

SUCCESS

And that includes item65.svg and item70.svg

--------------------
9) Tried 8 svg's

FAILED

As always, it stops on a random SVG, so I do not think the file is the issue.

This time it processed 7 out of 8 and died:

```
[+1194 ms] [+1266 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item20.svg' (1487ms)
[        ] [  +10 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item15.svg' (1493ms)
[        ] [        ] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item35.svg' (1492ms)
[  +91 ms] [  +21 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item65.svg' (1513ms)
[        ] [  +20 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item70.svg' (1533ms)
[        ] [        ] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item25.svg' (1536ms)
[        ] [   +3 ms] Finished transforming asset at path
'C:\Users\me\projectname\dev-projectname\assets\test\item5.svg' (1538ms)
[        ] [   +2 ms] Persisting file store
[        ] [  +18 ms] Done persisting file store
[        ] [  +11 ms] Target debug_android_application failed: Error: User-defined transformation of asset 
"C:\Users\me\projectname\dev-projectname\assets\test\item40.svg" failed.
[   +1 ms]            Transformer process terminated with non-zero exit code: 255
[        ]            Transformer package: vector_graphics_compiler
[        ]            Full command: C:\Flutter\bin\cache\dart-sdk\bin\dart.exe run vector_graphics_compiler
--input=C:\Users\me\AppData\Local\Temp\flutter_tools.d2ba8b6f\item40.svg-transformOutput0.svg
--output=C:\Users\me\AppData\Local\Temp\flutter_tools.d2ba8b6f\item40.svg-transformOutput1.svg
[        ]            stdout:
[        ]            stderr:
[        ]            Cannot rename file to
'C:\Users\me\projectname\dev-projectname\.dart_tool/pub\bin\vector_graphics_compiler\vector_graphics_compile
r.dart-3.5.3.snapshot', path =
'C:\Users\me\projectname\dev-projectname\.dart_tool/pub\bin\vector_graphics_compiler\tmpd270e2c4\vector_grap
hics_compiler.dart-3.5.3.snapshot.incremental.temp' (OS Error: Access is denied.
[        ]            , errno = 5)
[        ]            PathAccessException: Cannot rename file to
'C:\Users\me\projectname\dev-projectname\.dart_tool/pub\bin\vector_graphics_compiler\vector_graphics_compile
r.dart-3.5.3.snapshot', path =
'C:\Users\me\projectname\dev-projectname\.dart_tool/pub\bin\vector_graphics_compiler\tmpd270e2c4\vector_grap
hics_compiler.dart-3.5.3.snapshot.incremental.temp' (OS Error: Access is denied.
[        ]            , errno = 5)
[        ]            #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
[        ]            #1      copyAssets.<anonymous closure>
(package:flutter_tools/src/build_system/targets/assets.dart:157:19)
```

--------------------
10) Tried 7 items

FAILED

This time, it only finished 5 items out of 7.  

It is telling that the "finished" item count for a list of items (say 9 finished out of 10) can be HIGHER than the TOTAL number of items when it fails (for example, finished only 5 out of 7). 

Here is another clue:  When I include ALL my dirs and items, about 200 items, it processes MOST of them, then eventually fails. 

--------------------
11) Let's try 6 items:

FAILED

Processed 5 out of 6

--------------------
12) NOW, back to 5 items, which worked in the past

SUCCESS!

--------------------
13) As a final test, instead of processing a dir like /test in pubspec.yaml, I explicitly entered 12 separate files like:

```
    - path: assets/test/item10.svg
      transformers:
        - package: vector_graphics_compiler
```
        
FAILED

--------------------
So from my testing, > 5 items always fails.
